llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5026,6 +5006,9 @@ calculateRegisterUsage(VPlan &Plan, ArrayRef<ElementCount> VFs,
         for (auto *R : OpenIntervals) {
           if (isa<VPVectorPointerRecipe, VPVectorEndPointerRecipe>(R))
             continue;
+          if (auto *Phi = dyn_cast<VPReductionPHIRecipe>(R);
+              Phi && Phi->isInLoop())
+            continue;
```

Comment:
Why is this discarding the reduction PHI?

Also, if you rebase on main, you'll see this FIXME, that I think can now be addressed by this PR?
https://github.com/llvm/llvm-project/pull/126437/files#diff-da321d454a7246f8ae276bf1db2782bf26b5210b8133cb59e4d7fd45d0905decR5009-R5011